db/similarity.py

```diff
@@ -0,0 +1,173 @@
+from __future__ import absolute_import
+from flask import current_app
+
+import db
+from db.data import count_all_lowlevel
+from db.exceptions import NoDataFoundException, BadDataException
+import db.similarity_stats
+import similarity.metrics
+import similarity.utils
+
+from sqlalchemy import text
+
+
+def add_metrics(batch_size):
+    """Computes each metric from the similarity_metrics table for
+    each recording in the lowlevel table, in batches.
+
+    Args:
+        batch_size: the number of recordings to be inserted, per batch.
+    """
+    lowlevel_count = count_all_lowlevel()
+
+    with db.engine.connect() as connection:
+        metrics = similarity.utils.init_metrics()
+
+        # Collect and assign stats to metrics that require normalization
+        for metric in metrics:
+            db.similarity_stats.assign_stats(metric)
+
+        sim_count = count_similarity()
+        current_app.logger.info("Processed {} / {} ({:.3f}%)".format(sim_count,
```

Comment:
@alastair I looked for a solution to echo this information in the manage.py command rather than the db module but wasn't very successful. Since it all happens in one loop I'm not sure how we could send info back intermittently? Replaced with logger for now.